nvflare/fuel/f3/cellnet/utils.py

```diff
@@ -93,3 +99,38 @@ def decode_payload(message: Message, encoding_key=MessageHeaderKey.PAYLOAD_ENCOD
         # assume to be bytes
         pass
     message.remove_header(encoding_key)
+
+
+def format_size(num, binary=False):
+    """Format size in human-readable formats like  KB, MB, KiB, MiB
+
+    Args:
+        num: Size in bytes
+        binary: If binary, one K is 1024 bytes, otherwise 1000 bytes.
+
+    Returns: Size in human-readable format (like 10MB, 100.2GiB etc)
+
+    """
+
+    if binary:
+        kilo = 1024.0
+        suffix = "iB"
+    else:
+        kilo = 1000.0
+        suffix = "B"
+
+    unit_found = False
+    for unit in ("", "K", "M", "G", "T"):
+        if abs(num) < kilo:
+            unit_found = True
+            break
+        num /= kilo
+
+    if not unit_found:
+        unit = "P"
```

Comment:
A question regarding implementation, why "P" needs to be outside of the for loop?

Reply:
The loop can only go so far, we choose 'T' as last one in the loop. P is used for any larger numbers. So there must be a line to handle this. I added a test case to demonstrate this.